src/plugins/presentation_util/public/components/input_controls/control_types/options_list/options_list_component.tsx

```diff
@@ -0,0 +1,171 @@
+
+  useMount(() => {
+    typeaheadSubject
+      .pipe(
+        tap((rawSearchText) => setSearchString(rawSearchText)),
+        debounceTime(100)
+      )
+      .subscribe((search) => setDebouncedSearchString(search));
+    // default selections can be applied here...
+  });
+
+  const { indexPattern, timeRange, filters, field, query } = input;
+  useEffect(() => {
+    let canceled = false;
+    setIsLoading(true);
+    fetchData({
+      search: debouncedSearchString,
+      indexPattern,
+      timeRange,
+      filters,
+      field,
+      query,
+    }).then((newOptions) => {
+      if (canceled) return;
+      setIsLoading(false);
+      // We now have new 'availableOptions', we need to ensure the previously selected options are still selected.
+      const enabledIndices: number[] = [];
+      selectedOptions.current?.forEach((selectedOption) => {
+        const optionIndex = newOptions.findIndex(
+          (availableOption) => availableOption.label === selectedOption
+        );
+        if (optionIndex >= 0) enabledIndices.push(optionIndex);
+      });
+      newOptions = toggleAvailableOptions(enabledIndices, newOptions, true);
+      setAvailableOptions(newOptions);
+    });
+    return () => {
+      canceled = true;
+    };
+  }, [indexPattern, timeRange, filters, field, query, debouncedSearchString, fetchData]);
+
+  const updateItem = useCallback(
+    (index: number) => {
+      const item = availableOptions?.[index];
+      if (!item) return;
+
+      const toggleOff = availableOptions[index].checked === 'on';
+
+      const newAvailableOptions = toggleAvailableOptions([index], availableOptions, !toggleOff);
+      setAvailableOptions(newAvailableOptions);
+
+      if (toggleOff) selectedOptions.current.delete(item.label);
+      else selectedOptions.current.add(item.label);
+    },
+    [availableOptions]
+  );
+
+  const selectedOptionsString = Array.from(selectedOptions.current).join(
+    OptionsListStrings.summary.getSeparator()
+  );
+  const selectedOptionsLength = Array.from(selectedOptions.current).length;
+
+  const { twoLineLayout } = input;
+
+  const button = (
+    <EuiFilterButton
+      iconType="arrowDown"
+      className={classNames('optionsList--filterBtn', {
+        'optionsList--filterBtnSingle': !twoLineLayout,
+        'optionsList--filterBtnPlaceholder': !selectedOptionsLength,
+      })}
+      onClick={() => setIsPopoverOpen((openState) => !openState)}
+      isSelected={isPopoverOpen}
+      numFilters={availableOptions.length}
+      hasActiveFilters={selectedOptionsLength > 0}
+      numActiveFilters={selectedOptionsLength}
+    >
+      {!selectedOptionsLength ? OptionsListStrings.summary.getPlaceholder() : selectedOptionsString}
+    </EuiFilterButton>
+  );
+
+  return (
+    <EuiFilterGroup
+      className={classNames('optionsList--filterGroup', {
+        'optionsList--filterGroupSingle': !twoLineLayout,
+      })}
+    >
+      <EuiPopover
+        id="popoverExampleMultiSelect"
```

Comment:
Is this id correct? I can't find a reference to it in this PR

Reply:
Whoops, that's a leftover from copying from the EUI examples - will remove!